`qml.set_shots` is available (#7337)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

> * A new transfomer :func:`~.transforms.set_shots` has been added to set the number of shots for all QNodes in a pipeline.

Reviewer: ```suggestion
* A new QNode transform called :func:`~.transforms.set_shots` has been added to set or update the number of shots to be performed, overriding shots specified in the device.
```

>   This transform can be used to set the number of shots for all QNodes in a pipeline.

Reviewer: @JerryChen97 what do you mean by this? 🤔

>   This transform can be used to set the number of shots for all QNodes in a pipeline.

Author: ```suggestion
```

  [(#7337)](https://github.com/PennyLaneAI/pennylane/pull/7337)

>   ```python
>   device = qml.device("default.qubit", wires=2)
> 
>   @partial(set_shots, shots=1000)
>   @qml.qnode(device)
>   def circuit(x):
>       qml.RX(x, wires=0)
>       qml.RY(x, wires=1)
>       qml.CNOT(wires=[0, 1])
>       return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
>   ```

Reviewer: ````suggestion
  The :func:`~.transforms.set_shots` transform can be used as a decorator:

  ```python
  @partial(qml.set_shots, shots=2)
  @qml.qnode(qml.device(“default.qubit”, wires=1))
  def circuit():
      qml.RX(1.23, wires=0)
      return qml.sample(qml.Z(0))
  ```

  ```pycon
  >>> circuit()
  TODO
  ```
  
  Additionally, it can be used in-line to update a circuit's `shots`:

  ```pycon
  >>> new_circ = qml.set_shots(circuit, shots=(4, 10)) # shot vector
  TODO
  ```
````

Reviewer: @JerryChen97 could you fill in the TODOs? 😅 😁

Author: A slightly irrelevant but might be useful observation: your input method (or auto-reformatter?) seems to be generating something different @isaacdevlugt  :scream: 
![image](https://github.com/user-attachments/assets/40a3b386-48b4-4bcb-ad2a-b7f547ec30d9)

Author: I'll edit this along with TODO

<h3>New features since last release</h3>

* A new QNode transform called :func:`~.transforms.set_shots` has been added to set or update the number of shots to be performed, overriding shots specified in the device.
  [(#7337)](https://github.com/PennyLaneAI/pennylane/pull/7337)

  The :func:`~.transforms.set_shots` transform can be used as a decorator:

  ```python
  @partial(qml.set_shots, shots=2)
  @qml.qnode(qml.device("default.qubit", wires=1))
  def circuit():
      qml.RX(1.23, wires=0)
      return qml.sample(qml.Z(0))
  ```

  ```pycon
  >>> circuit()

>   [1. 1.]

Author: ```suggestion
  array([1., -1.])
```
